planner-api/src/main/java/pl/edu/agh/kuce/planner/balance/service/BalanceService.java

```diff
@@ -0,0 +1,28 @@
+package pl.edu.agh.kuce.planner.balance.service;
+
+import org.springframework.stereotype.Service;
+import pl.edu.agh.kuce.planner.auth.persistence.User;
+import pl.edu.agh.kuce.planner.balance.dto.BalanceDto;
+import pl.edu.agh.kuce.planner.balance.persistence.Balance;
+import pl.edu.agh.kuce.planner.balance.persistence.BalanceRepository;
+
+@Service
+public class BalanceService {
+    private final BalanceRepository balanceRepository;
+
+    public BalanceService(final BalanceRepository balanceRepository) {
+        this.balanceRepository = balanceRepository;
+    }
+
+    public void create(final User user, final Integer balance) {
+        balanceRepository.save(new Balance(user, balance));
+    }
+
+    public void update(final User user, final Integer newBalance) {
+        balanceRepository.updateBalance(user, newBalance);
+    }
+
+    public BalanceDto list(final User user) {
+        return new BalanceDto(balanceRepository.findByUser(user).get(0));
```

Comment:
If you expect a single entity to be returned from DB, you can declare returned value type of balanceRepository.findByUser(user) as Balance instead of List.

Reply:
I was thinking a little bit about subBalances, and because of that this look like that, but to be consistent I can change it to just Balance